File: src/main/java/com/trade/securities/adapter/in/web/EtnTradingInfoController.java
package com.trade.securities.adapter.in.web;

import com.trade.securities.application.port.in.GetEtnTradingInfoUseCase;
import com.trade.securities.domain.EtnTradingInfo;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/etn")
@RequiredArgsConstructor
public class EtnTradingInfoController {

    private final GetEtnTradingInfoUseCase getEtnTradingInfoUseCase;

    @GetMapping("/trading-info")
    public List<EtnTradingInfo> getEtnTradingInfo(@RequestParam String basDd) {
        return getEtnTradingInfoUseCase.getEtnTradingInfo(basDd);
    }
}

File: src/main/kotlin/com/trade/securities/adapter/in/web/EtnTradingInfoController.kt
package com.trade.securities.adapter.`in`.web

import com.trade.securities.application.port.`in`.GetEtnTradingInfoUseCase
import com.trade.securities.domain.EtnTradingInfo
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.QueryParam
import jakarta.ws.rs.core.MediaType

@ApplicationScoped
@Path("/api/etn")
@Produces(MediaType.APPLICATION_JSON)
class EtnTradingInfoController(
    private val getEtnTradingInfoUseCase: GetEtnTradingInfoUseCase
) {

    @GET
    @Path("/trading-info")
    fun getEtnTradingInfo(@QueryParam("basDd") basDd: String): List<EtnTradingInfo> {
        return getEtnTradingInfoUseCase.getEtnTradingInfo(basDd)
    }
}
